erpnext/hr/doctype/attendance_request/attendance_request.py

```diff
@@ -37,11 +37,14 @@ def create_attendance(self):
 				attendance.employee = self.employee
 				attendance.employee_name = self.employee_name
 				if self.half_day and date_diff(getdate(self.half_day_date), getdate(attendance_date)) == 0:
```

Comment:
```suggestion
				if self.half_day and getdate(self.half_day_date) == getdate(attendance_date):
```
Instead of computing `date_diff` you can simply check if those two dates are equal I think.